Mathlib/Tactic/ToAdditive.lean

```diff
@@ -540,62 +538,96 @@ def copyInstanceAttribute (src tgt : Name) : CoreM Unit := do
     trace[to_additive_detail] "Making {tgt} an instance with priority {prio}."
     addInstance tgt attr_kind prio |>.run'
 
-/-- A hack to add an attribute to a declaration.
-  We use the `missing` for the syntax, so this only works for certain attributes.
-  It seems to work for `refl`, `symm`, `trans`, `ext` and `coe`.
-  This does not work for most attributes where the syntax has optional arguments.
-  TODO: have a proper implementation once we have the infrastructure for this. -/
-def hackyAddAttribute (attrName declName : Name) (kind : AttributeKind := .global) :
-  CoreM Unit := do
-  let .ok attr := getAttributeImpl (← getEnv) attrName
-    | throwError "unknown attribute {attrName}"
-  attr.add declName .missing kind
-
-/-- Copy an attribute that stores enough information to test whether a declaration is in it
-  in a hacky way.
-  TODO: have a proper implementation once we have the infrastructure for this. -/
-def hackyCopyAttr [Inhabited β] (attr : SimpleScopedEnvExtension α β) (f : β → Name → Bool)
-  (attrName src tgt : Name) : CoreM Unit := do
-  if f (attr.getState (← getEnv)) src then
-    hackyAddAttribute attrName tgt
-
-/-- Copy attributes to the additive name. -/
-def copyAttributes (src tgt : Name) : CoreM Unit := do
-  -- Copy the standard `simp` attribute
-  copySimpAttribute src tgt
-  -- Copy the `norm_cast` attributes
-  copySimpAttribute src tgt pushCastExt
-  copySimpAttribute src tgt normCastExt.up
-  copySimpAttribute src tgt normCastExt.down
-  copySimpAttribute src tgt normCastExt.squash
-  -- copy `instance`
+/-- Warn the user when the multiplicative declaration has an attribute. -/
+def warnExt [Inhabited σ] (ext : PersistentEnvExtension α β σ) (f : σ → Name → Bool)
+  (thisAttr attrName src : Name) : CoreM Unit := do
+  if f (ext.getState (← getEnv)) src then
```

Comment:
This warning should be associated to a lint class and disablable, so that it is still possible to use `@[simp, to_additive]` on purpose.

Reply:
What is the best way to do this? Shall I do the following?
* Add an option `linter.toAdditiveWarnings` (where I can also add some other warnings)
* Instead of `logInfo` use "if option is true, then do `logLint ...`"
* never actually run `initialize addLinter ...`

Reply:
I'm also happy to do this after this gets merged, this is holding up quite a few PRs.

Reply:
yes to the first thing. For the second, why aren't the existing cases of deliberate `attr, to_additive` already making the build noisy?

Reply:
what cases? 
Also, I think that `logInfo` doesn't show up in CI...